app/models/tweet.rb

```diff
@@ -0,0 +1,9 @@
+class Tweet < ApplicationRecord
+  belongs_to :user, foreign_key: 'id_user'
+  has_many :bookmarks
+  has_many :likes
+  has_many :hashtags
+  belongs_to :retweeted_tweet, class_name: 'Tweet', foreign_key: 'retweet', optional: true
+  has_one :quote, foreign_key: 'id_tweet', primary_key: 'id_tweet'
```

Comment:
using a not defined column as a foreigh key, I'll recomend you to use either `quote_id` or `retweet_id` to reference the parent tweet